File: algorithms/search-algorithms/breadth-first-search/breadthFirstSearch.js
function breadthFirstSearch(rootNode, searchValue) {
  let queue = [rootNode];
  while (queue.length) {
    const node = queue[0];
    if (node.value && node.value === searchValue) {
      return true;
    } else if (node.left || node.right) {
      if (node.left) queue.push(node.left);
      if (node.right) queue.push(node.right);
    }
    queue.shift();
  }
  return false;
}

module.exports = breadthFirstSearch;

File: algorithms/search-algorithms/breadth-first-search/breadth_first_search.py
def breadth_first_search(root_node, search_value):
    queue = [root_node]
    while len(queue) > 0:
        node = queue[0]
        if node.value and node.value == search_value:
            return True
        elif node.left is not None or node.right is not None:
            if node.left is not None:
                queue.append(node.left)
            if node.right is not None:
                queue.append(node.right)
            queue = queue[1:]
    return False
